### ISSUE TITLE
Input of `LABEL:@-` no longer works

### ISSUE THREAD

**Issue by njdoyle** [ImageMagick/ImageMagick#1245]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am using the latest version of ImageMagick
- [x] I have searched [open](https://github.com/ImageMagick/ImageMagick/issues) and [closed](https://github.com/ImageMagick/ImageMagick/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
In a recent version of ImageMagick, using the command line option of `LABEL:@-` to create a label that reads text from STDIN no longer works.

I am able to create label images using this form `LABEL:'text'` but not the above form where text is read from STDIN.

### Steps to Reproduce
Running this command should produce `hello.png` which is an image that says "Hello World":
```
echo "Hello World" | magick LABEL:@- hello.png
```

Instead, it fails with this message:
```
magick: unable to read blob '-': No such file or directory @ error/blob.c/FileToBlob/1396.
```

### System Configuration

- ImageMagick version: `ImageMagick 7.0.8-9 Q16 x86_64 2018-08-07`
- Environment (Operating system, version and so on): MacOS 10.13.6
- Additional information:
```
$ magick -list configure

Path: /Users/nidoyle/Prefix/lib/ImageMagick-7.0.8//config-Q16HDRI/configure.xml

Name           Value
-------------------------------------------------------------------------------
CC             gcc
CFLAGS         -I/Users/nidoyle/Prefix/include/libxml2 -I/Users/nidoyle/Prefix/include -I/Users/nidoyle/Prefix/include -I/Users/nidoyle/Prefix/include/libpng16 -I/Users/nidoyle/Prefix/include -I/Users/nidoyle/Prefix/include/openjpeg-2.3 -I/Users/nidoyle/Prefix/include -I/Users/nidoyle/Prefix/include/freetype2 -I/Users/nidoyle/Prefix/include/libpng16 -I/Users/nidoyle/Prefix/include -I/Users/nidoyle/Prefix/include   -g -O2 -Wall -mtune=native -fexceptions -D_FORTIFY_SOURCE=0 -pthread -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16
CODER_PATH     /Users/nidoyle/Prefix/lib/ImageMagick-7.0.8/modules-Q16HDRI/coders
CONFIGURE      ./configure  '--prefix=/Users/nidoyle/Prefix' '--with-gcc-arch=native' '--disable-openmp' '--without-x' 'LDFLAGS=-L/Users/nidoyle/Prefix/lib -Wl,-rpath,/Users/nidoyle/Prefix/lib ' 'CPPFLAGS=-I/Users/nidoyle/Prefix/include '
CONFIGURE_PATH /Users/nidoyle/Prefix/etc/ImageMagick-7/
COPYRIGHT      Copyright (C) 1999-2018 ImageMagick Studio LLC
CPPFLAGS       -I/Users/nidoyle/Prefix/include/ImageMagick-7
CXX            g++
CXXFLAGS       -g -O2 -pthread
DEFS           -DHAVE_CONFIG_H
DELEGATES      bzlib mpeg freetype jng jpeg lcms openjp2 png tiff webp xml zlib
DISTCHECK_CONFIG_FLAGS 'CPPFLAGS=-I/Users/nidoyle/Prefix/include -I/Users/nidoyle/Prefix/include ' 'LDFLAGS=-L/Users/nidoyle/Prefix/lib -Wl,-rpath,/Users/nidoyle/Prefix/lib -L/Users/nidoyle/Prefix/lib -Wl,-rpath,/Users/nidoyle/Prefix/lib '  --disable-deprecated  --with-quantum-depth=16  --with-jemalloc=no  --with-umem=no  --with-autotrace=no  --with-gslib=no  --with-fontpath=  --with-rsvg=no  --with-perl=no
DOCUMENTATION_PATH /Users/nidoyle/Prefix/share/doc/ImageMagick-7
EXEC-PREFIX    /Users/nidoyle/Prefix
EXECUTABLE_PATH /Users/nidoyle/Prefix/bin
FEATURES       DPC HDRI Cipher
FILTER_PATH    /Users/nidoyle/Prefix/lib/ImageMagick-7.0.8/modules-Q16HDRI/filters
GIT_REVISION   14617
HOST           x86_64-apple-darwin17.6.0
INCLUDE_PATH   /Users/nidoyle/Prefix/include/ImageMagick-7
LDFLAGS        -L/Users/nidoyle/Prefix/lib -L/Users/nidoyle/Prefix/lib -Wl,-rpath,/Users/nidoyle/Prefix/lib -L/Users/nidoyle/Prefix/lib -Wl,-rpath,/Users/nidoyle/Prefix/lib
LIB_VERSION    0x708
LIB_VERSION_NUMBER 7,0,8,9
LIBRARY_PATH   /Users/nidoyle/Prefix/lib/ImageMagick-7.0.8
LIBS             -L/Users/nidoyle/Prefix/lib -llcms2 -ltiff -L/Users/nidoyle/Prefix/lib -lfreetype  -ljpeg   -L/Users/nidoyle/Prefix/lib -lpng16 -lz        -L/Users/nidoyle/Prefix/lib -lwebpmux -lwebp -L/Users/nidoyle/Prefix/lib -lwebp        -lbz2  -L/Users/nidoyle/Prefix/lib -lopenjp2    -L/Users/nidoyle/Prefix/lib -lxml2  -L/Users/nidoyle/Prefix/lib -lz    -lm
NAME           ImageMagick
PCFLAGS        -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16
PREFIX         /Users/nidoyle/Prefix
QuantumDepth   16
RELEASE_DATE   2018-08-07
SHARE_PATH     /Users/nidoyle/Prefix/share/ImageMagick-7
SHAREARCH_PATH /Users/nidoyle/Prefix/lib/ImageMagick-7.0.8/config-Q16HDRI
TARGET_CPU     x86_64
TARGET_OS      darwin17.6.0
TARGET_VENDOR  apple
VERSION        7.0.8
WEBSITE        http://www.imagemagick.org

Path: [built-in]

Name           Value
-------------------------------------------------------------------------------
FEATURES
NAME           ImageMagick
QuantumDepth   16
```

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in GIT master branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://www.imagemagick.org/download/beta/ by sometime tomorrow.

**Comment by njdoyle** [comment]
Confirmed this is fixed in 7.0.8-10. Thank you!
